staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/aggregator.go

```diff
@@ -59,12 +59,24 @@ func mergeSpec(dest, source *spec.Swagger) {
 		if dest.Definitions == nil {
 			dest.Definitions = spec.Definitions{}
 		}
-		dest.Definitions[k] = v
+		// We don't allow CRDs to have name conflict, but currently there is one corner
+		// case that a CRD can define apiextensions.apis.pkg.apiextensions-apiserver.k8s.io
+		// customresourcedefinitions, which causes definition conflict against
+		// apiextensions-apiserver's static spec. We skip the definition if it's already there
+		if _, found := dest.Definitions[k]; !found {
+			dest.Definitions[k] = v
+		}
 	}
 	for k, v := range source.Paths.Paths {
 		if dest.Paths.Paths == nil {
 			dest.Paths.Paths = map[string]spec.PathItem{}
 		}
-		dest.Paths.Paths[k] = v
+		// We don't allow CRDs to have name conflict, but currently there is one corner
+		// case that a CRD can define apiextensions.k8s.io customresourcedefinitions, which
+		// causes path conflict against apiextensions-apiserver's static spec. We skip
+		// the path if it's already there
+		if _, found := dest.Paths.Paths[k]; !found {
+			dest.Paths.Paths[k] = v
```

Comment:
Would not mention CRDs here. Make it clear in the func description that existing paths are not overwritten. I.e. the first spec (with highest priority) wins with its paths.

Reply:
maybe rename it to `mergeSpecIgnoringExistingPaths`.